components/nimbus/src/tests/test_evaluator.rs

```diff
@@ -23,12 +23,8 @@ fn test_locale_substring() -> Result<()> {
         locale: Some("de-US".to_string()),
         ..Default::default()
     };
-    let targeting_attributes = ctx.into();
-    let event_store = Arc::new(Mutex::new(EventStore::new()));
-    assert_eq!(
-        targeting(expression_statement, &targeting_attributes, event_store),
-        None
-    );
+    let th = ctx.into();
+    assert_eq!(targeting(expression_statement, &th), None);
```

Comment:
Okay so if I'm understanding this correctly, the `AppContext` type is able to `.into()` on `TargetingHelper` types because `new` on the `TargetingHelper` has the generic `<C: Serialize>`, and `AppContext` is serializable.

One thing I'm not tracking here is how it knows how to instantiate the EventStore when used in this way? Does it effectively use `Default::default()` for `Arc<Mutex<EventStore>>`?

Reply:
There are [several `impl`s of `From` and `Default` used in testing in `test_enrollment.rs`, here](https://github.com/mozilla/application-services/blob/32de3d621b7f0662d0b5da5bbb700e1c251c35e7/components/nimbus/src/tests/test_enrollment.rs#L1199-L1226).